runelite-client/src/main/java/net/runelite/client/plugins/specialcounter/SpecialCounterPlugin.java

```diff
@@ -246,4 +268,19 @@ private int getHit(SpecialWeapon specialWeapon, int deltaExperience)
 			return (int) Math.round(damageOutput);
 		}
 	}
+
+	private void notifySpecialChange(int special)
```

Comment:
notifySpecialChange might be a bit of a deceiving name as that is not always what it does. I think naming it something like checkFullSpecial.